### PR TITLE
fix(quantic): es build plugin added to allow rtk query to work inside salesforce

### PR DESCRIPTION
**Pull request by mmitiche**
### The issue:
When submitting an evaluation  from the Quantic Generated Answer in Salesforce the following issue occur preventing a successful request:

```
    at headless.js:30:99706
headless.js:18 TypeError: Failed to convert value to 'AbortSignal'.
    at https://data-ruby-6040-dev-ed.scratch.my.site.com/examples/resource/1732042845000/coveoheadless/headless.js:30:99706 Action dispatch error answer/executeMutation/rejected 
{type: 'answer/executeMutation/rejected', payload: undefined, meta: {…}, error: {…}}
error: 
message: "Failed to construct 'Request': Failed to read the 'signal' property from 'RequestInit': Failed to convert value to 'AbortSignal'."
name: "TypeError"stack: 
"TypeError: Failed to convert value to 'AbortSignal'.\n    at https://data-ruby-6040-dev-ed.scratch.my.site.com/examples/resource/1732042845000/coveoheadless/headless.js:30:99706"
"answer/executeMutation/rejected"
``` 

This is occurring because in Salesforce the AbortController/AbortSignal web API are not available and supported, so we are using a pony-fill that we are injecting in the headless bundle built for Quantic and Salesforce.

The problem is that Redux Toolkit Query does not seem to like this ponyfill and throws an error when starting to deal with it:

https://github.com/user-attachments/assets/e659536d-1f61-47e3-9c81-3378a7f2597f

To address this issue, we can avoid passing the Abort Signal to `fetchBaseQuery` which is the function  provided by RTK Query and that is executed to fetch for the Answer API in `answer-slice.ts` .
But to make sure this fix only impacts Salesforce and the Headless bundle used for Quantic, I added an es build plugin that will do a small modification to the `answer-slice.ts`  when building, instead of calling `fetchBaseQuery` like this:
```
    const data = fetchBaseQuery({
      baseUrl: `${platformEndpoint}/rest/organizations/${organizationId}/answer/v1/configs/${answerConfigurationId}`,
    })(updatedArgs, api, extraOptions);
```
it will be called like the following:
```
    const data = fetchBaseQuery({
      baseUrl: `${platformEndpoint}/rest/organizations/${organizationId}/answer/v1/configs/${answerConfigurationId}`,
    })(updatedArgs, {...api, signal: null}, extraOptions);
```

to allow properly submitting evaluations in Salesforce:

https://github.com/user-attachments/assets/8fd7ef9c-bc47-4fc0-b6be-f6cfa1f1072e

[SFINT-5831]: https://coveord.atlassian.net/browse/SFINT-5831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/quanticFeedbackModalQna.js [hunk 1]
```diff
@@ -168,7 +168,7 @@ export default class QuanticFeedbackModalQna extends LightningModal {
       return;
     }
     this.handleSubmit({
-      correctTopicValue: this.correctTopicValue,
+      correctTopic: this.correctTopicValue,
```
Comment (alexprudhomme) [inline packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/quanticFeedbackModalQna.js:171]: What is that change ? Seems unrelated to me

**Review by alexprudhomme — APPROVED** [review]
Thanks for the great explanation

**Review by SimonMilord — APPROVED** [review]
Nice!

File: packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/quanticFeedbackModalQna.js (hunk 1 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/quanticFeedbackModalQna.js:171]: Ah yeah forgot to mention that in pr description,
I also noticed that we were passing a wrong key when passing the parameter to submit feedback from Quantic,
That's the expected type of this parameter:

Screenshot 2024-11-28 at 11 01 59 AM

**Review by dmgauthier — APPROVED** [review]

**Comment by dmgauthier** [comment]
Very wise!
